Document positions file usage in loki.source.docker.md (#4686)
grafana/agent: docs/sources/flow/reference/components/loki.source.docker.md

* `loki_source_docker_target_entries_total` (gauge): Total number of successful entries sent to the Docker target.
* `loki_source_docker_target_parsing_errors_total` (gauge): Total number of parsing errors while receiving Docker messages.

## Component behavior
The component uses its data path (a directory named after the domain's

> fully qualified name) to store its _positions file_. The positions file is used
> to store read offsets, so that in case of a component or Agent restart,

Reviewer: ```suggestion
fully qualified name) to store its _positions file_. The positions file
stores the read offsets so that if there is a component or Agent restart,
```
